cogs/cog_timer.py

```diff
@@ -13,13 +12,33 @@ def __init__(self, bot):
 
     @commands.command(name="set-timer", aliases=['st'])
     async def set_timer(self, ctx, time: str, *msg: str):
-        """ Creates a Timer instance. """
+        """
+        Creates a Timer instance.
+        Note: 'time' must be in range [1 .. 120]
+        """
 
-        userid = ctx.author.id
+        user = ctx.author
+
+        # Verify time inputted is valid
+        try:
+            time = int(time)
+        except ValueError:  # not an integer
+            await ctx.send(embed=discord.Embed(
+                description="Non-numeric timer duration given!",
+                colour=cfg.colors.ERROR
+            ))
+            return
+
+        if not 1 <= time <= 120:  # not in allowed range of integers
+            await ctx.send(embed=discord.Embed(
+                description="Timer duration not in acceptable range! [1 .. 120]",
+                colour=cfg.colors.ERROR
+            ))
+            return
 
         # Timer creation
         msg = ' '.join(msg)
-        timer_obj = timer.Timer(userid=userid, td_secs=int(time) * 60, msg=msg, discord_message=ctx.message)
+        timer_obj = timer.Timer(userid=user.id, td_secs=int(time) * 60, msg=msg, discord_message=ctx.message)
```

Comment:
Redundant cast: we've already casted time to an int, but now we're doing it again